src/Agent.Worker/Program.cs

```diff
@@ -1,40 +1,65 @@
-﻿using System;
+﻿using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
 using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Microsoft.VisualStudio.Services.Agent.Worker
 {
     public static class Program
     {
-        public static Int32 Main(string[] args)
+        public static int Main(string[] args)
         {
-            int resultCode = 0;
-            using (HostContext hc = new HostContext("Worker"))
+            var tokenSource = new CancellationTokenSource();
+            var hc = new HostContext("Worker", tokenSource.Token);
+            return RunAsync(args, hc, tokenSource).GetAwaiter().GetResult();
```

Comment:
Beyond this line, the program is testable :)

Again, there was some change here since I removed the cancellation token *source* from the host context.